src/lib/hasFieldSelection.ts

```diff
@@ -102,18 +102,18 @@ export const parseConnectionArgsFromConnection = (
                       arg.name.value
                     )
                   ) {
-                    let val =
-                      (arg as any).value.value ||
-                      info.variableValues[arg.name.value] ||
-                      0
+                    let val: string
 
-                    // Emission sends back `count` and `cursor` as the
-                    // variables for `first` and `after`.
-                    // TODO: Improve introspection/visiting so we don't need this.
-                    if (arg.name.value === "first" && !val) {
-                      val = info.variableValues["count"]
-                    } else if (arg.name.value === "after" && !val) {
-                      val = info.variableValues["cursor"]
+                    if (arg.value.kind === "Variable") {
+                      const variableName = arg.value.name.value
+                      val = info.variableValues[variableName]
+                    } else if (
+                      arg.value.kind === "IntValue" ||
+                      arg.value.kind === "StringValue"
+                    ) {
+                      val = arg.value.value
+                    } else {
+                      val = "0"
                     }
 
                     connectionArgs[arg.name.value] = parseInt(val) || val
```

Comment:
I’m surprised TS isn’t complaining about this, but iirc `parseInt` should really always take its second argument, to indicate the value should be parsed as 10-base

Reply:
I have seen that complaint before! Odd.